### ISSUE TITLE
Update initial upload of bavaria-quebec dataset

### ISSUE THREAD

**Issue by mathieuboudreau** [neuropoly/data-management#180]
After the initial upload of `data.neuro.polymtl.ca:datasets/bavaria-quebec-spine-ms`, @jcohenadad provided the following suggestions:

* [x] (minor) move everything under "20221012_NeuroPoly_Cohort_v1" directly at the root (convention for BIDS datasets-- all subjects should be at the root)
* [x] (minor) the README to use is already inside 20221012_NeuroPoly_Cohort_v1
* [x] (major) the labels should be under 'derivatives/'. See discussion: https://github.com/sct-pipeline/bavaria-quebec/issues/13
* [ ] include the preprocessing script @Sandrine Bédard

I've just created a branch `mb/20221021_update` and committed applied all the updates which were transferred to me by Julian McGinnis, except for the last point (Julian's upload did not include Sandrine's scripts).

Further updates should be coming, because @mguaypaq found that the dataset does not currently pass the bids validator.

**Comment by mguaypaq** [comment]
@mathieuboudreau I think there are some files missing in the most recent upload, namely all the `*.nii.gz` files under `derivatives/`. (Given that the dataset is private but that this issue is publicly visible, I'm hesitant to post too many details.)

I thought I could fix this myself, because I thought they were just moved/renamed versions of the old `*_dseg.nii.gz` files, but according to git-annex the contents do not match (they have different hashes), so I'll need you to upload the new files.

Right now, git-annex knows the hashes of the files, but not the contents. Possibly this would be fixed by running the following commands in your copy of the repository (taken from [the git-annex upload instructions on the lab wiki](https://intranet.neuro.polymtl.ca/computing-resources/data/git-datasets.html#upload)):
```
git annex copy --to=origin
git annex sync --no-content --only-annex
```

Once this is done, the following command should report that each file has 1 copy available on the server:
```
git annex whereis derivatives/
```

My understanding is also that all the `*_dseg.nii.gz` files that are **outside** of `derivatives/` should be gone, but since the contents don't match the files that are now **inside** `derivatives/` I'm not sure whether I should remove them.

**Comment by mathieuboudreau** [comment]
>git annex copy --to=origin
>git annex sync --no-content --only-annex

Ahh thanks for catching that @mguaypaq , I did indeed accidentally skip this and pushed without doing so. I've just done so, and the `whereis` returns both locations (data server and local).

**Comment by mguaypaq** [comment]
I can now `git annex get` the image files under `derivatives/`, thanks! In fact, they're listed as available both from the server and from your local copy (since the version of `git annex` on our servers is sadly slightly too old to recognize the `annex.private` configuration option).

- [ ] Note to future self: drop the references to @mathieuboudreau's local copy of the data once he gets rid of it (using `git annex dead 3c33e0bb-938e-400a-8dcf-cb17f8b74d16` and possibly `git annex forget --drop-dead`).

Also, now that I have access to the files under `derivatives/`, I can see that they're only different from the corresponding files outside `derivatives/` in a superficial way: the compressed `.nii.gz` files are different, but the uncompressed `.nii` files they contain are identical. So it must be just some gzip metadata that's different, like a timestamp or something. Verified with the following bash command, if anyone is curious:

```bash
for file in sub-*/ses-*/anat/*_acq-ax_dseg.nii.gz; do
if cmp <(zcat "$file") <(zcat "derivatives/${file/_dseg./_lesion-manual.}")
then echo "same for ${file%%/*}"
else echo "difference in ${file%%/*}"
fi done
```

**Comment by mguaypaq** [comment]
I pushed some changes to `master` on the git-annex server:
* Delete duplicates of manual segmentation files from outside `derivatives/`
* Fix typos in README
* Add bids-validator configuration file, ignoring:
  * INCONSISTENT_PARAMETERS, because of a few nifti files with different dimensions and/or spatial resolution.
  * MISSING_SESSION, because we're ok with using different session labels for each subject.

What's missing now (that I can't do myself), is the preprocessing script, and fixing the units (seconds vs milliseconds) in the RepetitionTime and EchoTime fields.

**Comment by jqmcginnis** [comment]
@mguaypaq @mathieuboudreau I just updated the dataset, addressing the shortcomings (c.f. listed issues) and adding spinal cord segmentation files. For the link to the download please look at the channel in slack. If any issues arise, feel free to add them here again!

**Comment by jqmcginnis** [comment]
As we provide the spinal cord segmentation files (as manual segmentations) now, @sandrinebedard and I decided to omit a preprocessing script for now.

**Comment by mathieuboudreau** [comment]
@mguaypaq Julian's updated dataset has been uploaded to a new branch, `mb/20221102_update`. The branch was created from `master`, and I updated it prior to making the branch. I then deleted the old `derivatives` folder, and copied Julian's uploaded directory tree.

Hopefully this doesn't overwrite any of the changes you had made to `master`. Please have a look and let us know if it is mergeable in the current state. Thanks!

**Comment by mguaypaq** [comment]
I was able to merge the changes, and I pushed the new version to `master` on the server.

The `bids-validator` still complains that `RepetitionTime` and `EchoTime` are expressed in milliseconds, rather than in seconds, in violation of the BIDS specification. Is this something we're going to fix? I can write a script to do it, I just need to know that it's ok.

A small detail: I noticed that all the dates in the json files under `derivatives/labels/` are in the format `08.02.2022` (so, `dd.mm.yyyy`), whereas the format `2022-02-08` is more in line with the BIDS standard (see [date-formats](https://bids-specification.readthedocs.io/en/stable/glossary.html#date-formats) in the spec). Do we care about this? Again, I can write a script to do it if we want to make this change.

**Comment by jqmcginnis** [comment]
@mguaypaq thanks for bringing this to my attention again! :slightly_smiling_face: The thing is, we currently have an in-house pipeline that generates the NIFTI from DICOM and also generates the jsons with the wrong conventions for `RepetitionTime` and `EchoTime` and generates the date tags in the unconventional format (that I actually was not aware of - thanks!). I wanted to talk to the programmer of that pipeline and the medical doctors if they could incorporate these changes already into their pipeline (thus there would be no need for us to change anything). For our collaboration we can also do this via a post-processing script, however this will also introduce coexistence of different conventions into our database, that's why I was refraining / postponing any changes w.r.t. to the issues you stated. 

I am also thinking about introducing git-annex to our database, perhaps we can discuss this briefly in our upcoming meeting with Julien?

Thanks for offering your help with this!

**Comment by mguaypaq** [comment]
Not to my knowledge. @jqmcginnis, has the "upcoming meeting" that you mentioned happened yet? (I'm not aware of which meeting this would be, myself.)

**Comment by jqmcginnis** [comment]
@mathieuboudreau @mguaypaq sorry for the delay - last week's meeting got postponed to tomorrow (Nov 16), I will definitely address this in the meeting and let you know as soon as possible. Thanks for the follow up!

**Comment by mguaypaq** [comment]
@jqmcginnis, I'm just going through my old assigned issues that are still open, is there any update on this one?

**Comment by jqmcginnis** [comment]
@mguaypaq 

Good news - we have a revised cohort here. Short summary of changes:

1. Removal of ssome patients due to qualiyt issues (removed entirely, they are no longer included in the new BIDS.
2. Fixes with respect to naming - all mask should have the appropriate BIDS names, e.g. sub-m123456_ses-20121314_seg-manual_T2w.nii.gz and sub-m123456_ses-20121314_lesions-manual_T2w.nii.gz 
3. Fixed the Echo / Repetition Time Issues
4. Changed the date descriptor in the json files
5. updated the CHANGES file, version update to 1.1.0

Please let me know if there any issues I forgot to address or arise from the update.

I will ping you via NeuroPoly slack to start the file transfer.

**Comment by mguaypaq** [comment]
- The file transfer worked,
- I imported the files into `data.neuro.polymtl.ca:datasets/bavaria-quebec-spine-ms`,
- `bids-validator .` is happy, and
- `git annex get` on a fresh clone of the repository works.

So, everything looks good, thanks!
